### PR TITLE
bugfix(modflow.py): need to specify delc for pixel width and rot angle

### PR DESCRIPTION
**Pull request by mnfienen**
Two small things for getting affine from `modelgrid.json`:

First, if the a parameter is the width of a pixel, this should be `delc` rather than `delr`.

Second, I got goofy results on a rotated grid relative to a transform made using `modflow-export` using`MFexportGrid`. The difference was the `b` and `e` had opposite sign. Flipping the sign in the `Affine().rotation(angle)` call fixed this. However, I realize this is inconsistent with the docstring here: https://github.com/rasterio/affine/blob/master/affine/tests/test_rotation.py. Maybe the modegrid.json convention is reversed? In any case, making this switch resulted in proper results for this rotated model (18 degrees anticlockwise) and not reversing the angle resulted in negative values for `r` in the `get_ij` funciton.

### CODE REVIEW

**Comment by aleaf** [comment]
Hey @mnfienen thanks for taking a look at this. 

TL;DR- are you sure that delc/delr are consistent on your end, or is there an issue with how Modflow-setup is writing them to the grid.json file? We want to keep delc/delr as-is, but I think you're right that the rotation angle needs to be inverted.

Re delc/delr: I believe the a parameter is spacing in the x direction, which would correspond to delr (spacing _along_ a row). Maybe confusing, but that's how Flopy (and maybe Modflow?) does it, so I follow that convention for Modflow-setup to avoid confusion, especially in working with the DIS package and model grid object.

Re: rotation- I think we follow the Flopy convention for the direction of angrot (positive counter clockwise for an origin in the lower left corner). The [Flopy grid docstring unfortunately doesn't say anything about direction](https://github.com/modflowpy/flopy/blob/c253bab45409d2c01705494aef2b8532e6da08a0/flopy/discretization/grid.py#L25), but I think I'd have to be the same, otherwise we'd run into trouble with the subclassed grid object in Modflow-setup. 

I think maybe the confusion is in what origin is used. In Modflow-setup, the ``MFsetupGrid`` object accepts either the lower left or upper left corner to define the grid location, but the rotation direction (positive CCW) is always about the lower left corner. In contrast, the ``affine.Affine`` instance for the ``transform`` attribute is always defined based on the upper left corner (hence the negative y-spacing), so we have to flip the rotation sign to get rotation in the same direction. For example,  consider a simple grid with the lower left at 0, 0 and a 45 degree rotation ccw.
```python
from affine import Affine
trans_ll = Affine(1, 0, 0,
                  0, 1, 0)
```
the upper left corner is:
```python
trans_ll * Affine.rotation(45) * (0, 1)
(-0.7071067811865475, 0.7071067811865476)
```
we can redefine that with a new Affine object with the same upper left:
```python
trans_ul = Affine(1, 0, -0.707,
                  0, -1, 0.707)
```
to get the same lower left as the first instance, we need to flip the rotation sign
```python
trans_ul * Affine.rotation(-45) * (0, 1)
(0.00010678118654772195, -0.0001067811865474999)
```

**Comment by mnfienen** [comment]
hey @aleaf  - oh man, I'm always reverseing `delr` and `delc` in MODFLOW! The original issue was that both values were `delr` but I agree I changed the wrong one. I will fix.

for the rotation issue, you must have gone down this road before in `modflow-export` because here's how the transform is calculated there - on the `MFexportGrid` object:
```python
@property
def transform(self):
    """Rasterio Affine object (same as transform attribute of rasters).
    """
    return Affine(self.delr[0], 0., self.xul,
                  0., -self.delc[0], self.yul) * \
           Affine.rotation(angle=-self.angrot)
```
Note the negative angle specification - the results from `modflow-export` are oriented correctly, so one way or another, I think we need the negative angle. Calling this, the angle was positive and meant to be anticlockwise, so should be consistent. Also, I see I did reverse `delr` and `delc` as you noticed. I commited that correction
